### PR TITLE
Use feast core images from GCR instead of building them ourselves

### PR DESCRIPTION
**Pull request by pyalex**
Signed-off-by: Oleksii Moskalenko <moskalenko.alexey@gmail.com>

**What this PR does / why we need it**:

In this PR we install main components of Feast through public helm chart and not compiling them from submodule for e2e tests. Thus, submodule can be deleted.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oavdeev** [comment]
Would that mean that e2e tests would run against whatever is in master branch of core/serving at the time of the run?

**Comment by pyalex** [comment]
@oavdeev the intention is to use `latest` tag, not master. And eventually get rid of git submodule.

File: .github/workflows/complete.yml [hunk 1]
```diff
@@ -17,10 +17,16 @@ jobs:
...
           submodules: recursive
+      - uses: actions/checkout@v2
+      - uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
       - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
         with:
           version: '290.0.1'
           export_default_credentials: true
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
```
Comment (woop) [inline .github/workflows/complete.yml:28]: Which GCP_PROJECT_ID is this using, and dont we need builds for PRs?

File: infra/scripts/k8s-common-functions.sh [hunk 2]
```diff
@@ -65,20 +65,25 @@ function helm_install {
...
-        --timeout 15m \
-        --set "feast-jupyter.image.repository=${DOCKER_REPOSITORY}/feast-jupyter" \
-        --set "feast-jupyter.image.tag=${JUPYTER_GIT_TAG:-$GIT_TAG}" \
-        --set "feast-online-serving.image.repository=${DOCKER_REPOSITORY}/feast-serving" \
-        --set "feast-online-serving.image.tag=${SERVING_GIT_TAG:-$GIT_TAG}" \
-        --set "feast-jobservice.image.repository=${DOCKER_REPOSITORY}/feast-jobservice" \
-        --set "feast-jobservice.image.tag=${JOBSERVICE_GIT_TAG:-$GIT_TAG}" \
-        --set "feast-core.image.repository=${DOCKER_REPOSITORY}/feast-core" \
-        --set "feast-core.image.tag=${CORE_GIT_TAG:-$GIT_TAG}" \
+    helm repo add feast-charts https://feast-charts.storage.googleapis.com
```
Comment (woop) [inline infra/scripts/k8s-common-functions.sh:68]: @pyalex going forward, do you think its better to depend on the helm chart repository on GCS (where we can have specific versions, but can possibly push master), or do you want to use a submodule of the helm chart repo? I am assuming the later, right?

File: .github/workflows/complete.yml (hunk 1 above)
Comment (pyalex) [reply .github/workflows/complete.yml:28]: this is `kf-feast`. For PRs we still use hosted runner. At least I don't see a lot of alternatives if we want to continue to use prow

File: infra/scripts/k8s-common-functions.sh (hunk 2 above)
Comment (pyalex) [reply infra/scripts/k8s-common-functions.sh:76]: I'm totally fine with using public helm chart here (from GCS)

**Pull request merged**